src/position-manager/PositionManagerBase.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: UNLICENSED
+// Copyright (c) 2025 Aave Labs
+pragma solidity 0.8.28;
+
+import {IERC20} from 'src/dependencies/openzeppelin/IERC20.sol';
+import {IERC20Permit} from 'src/dependencies/openzeppelin/IERC20Permit.sol';
+import {Multicall} from 'src/utils/Multicall.sol';
+import {EIP712Types} from 'src/libraries/types/EIP712Types.sol';
+import {ISpoke} from 'src/spoke/interfaces/ISpoke.sol';
+import {IPositionManagerBase} from 'src/position-manager/interfaces/IPositionManagerBase.sol';
+
+/// @title PositionManagerBase
+/// @author Aave Labs
+/// @notice Base implementation for position manager common functionalities.
+abstract contract PositionManagerBase is IPositionManagerBase, Multicall {
+  /// @inheritdoc IPositionManagerBase
+  address public immutable override SPOKE;
+
+  /// @dev Constructor.
+  /// @param spoke_ The address of the spoke contract.
+  constructor(address spoke_) {
+    require(spoke_ != address(0), InvalidAddress());
+    SPOKE = spoke_;
+  }
+
+  /// @inheritdoc IPositionManagerBase
+  function setSelfAsUserPositionManagerWithSig(
+    address user,
+    bool approve,
+    uint256 nonce,
+    uint256 deadline,
+    bytes calldata signature
+  ) external {
+    try
+      ISpoke(SPOKE).setUserPositionManagerWithSig({
+        positionManager: address(this),
+        user: user,
+        approve: approve,
+        nonce: nonce,
+        deadline: deadline,
+        signature: signature
+      })
+    {} catch {}
+  }
+
+  /// @inheritdoc IPositionManagerBase
+  function permitReserve(
```

Comment:
`permitReserveUnderlying`?

Reply:
Tho I can see why, since on the Spoke the Reserve is a defined struct, here we only care about the underlying of the given reserve.